networklibrary/src/main/java/com/eis/smsnetwork/SMSNetDictionary.java

```diff
@@ -46,33 +53,120 @@ public void removeResource(String key) {
+     */
+    public void removeResourceFromSMS(@NonNull String key) {
+        removeResource(removeEscapes(key));
+    }
+
+    /**
+     * Returns a String containing all keys and resources from the dictionary, in the format
+     * "key1 resource1 key2 resource2", after calling {@link SMSNetDictionary#addEscapes(String)} on each of them.
+     *
+     * @return All keys and resources present in the dictionary, ready to be sent through an SMS.
+     * If there are none, returns null.
+     */
+    public String getAllKeyResourcePairsForSMS() {
+        Set<String> keySet = dict.keySet();
+        ArrayList<String> keys = new ArrayList<>(keySet.size());
+        ArrayList<String> resources = new ArrayList<>(keySet.size());
+        for (String key : keySet) {
+            String resource = getResource(key);
+            resources.add(addEscapes(resource));
+            keys.add(addEscapes(key));
+        }
+        StringBuilder result = new StringBuilder();
+        for (int i = 0; i < keys.size(); i++) {
+            result.append(keys.get(i)).append(BroadcastReceiver.FIELD_SEPARATOR);
+            result.append(resources.get(i)).append(BroadcastReceiver.FIELD_SEPARATOR);
+        }
+        try {
+            return result.deleteCharAt(result.length() - 1).toString();
+        } catch (StringIndexOutOfBoundsException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Adds a backslash before every character corresponding to
+     * {@link com.eis.smsnetwork.broadcast.BroadcastReceiver#FIELD_SEPARATOR}. Needed when reading
+     * keys and resources before sending them through an SMS.
+     *
+     * @param string The String where to escape every occurrence of
+     *               {@link com.eis.smsnetwork.broadcast.BroadcastReceiver#FIELD_SEPARATOR}.
+     * @return A String where every
+     * {@link com.eis.smsnetwork.broadcast.BroadcastReceiver#FIELD_SEPARATOR} is escaped with a
+     * backslash.
+     */
+    public static String addEscapes(@NonNull String string) {
+        String replacement = Matcher.quoteReplacement("\\" + BroadcastReceiver.FIELD_SEPARATOR);
+        return string.replaceAll(BroadcastReceiver.FIELD_SEPARATOR, replacement);
+    }
+
+    /**
+     * Whenever a backslash precedes a {@link BroadcastReceiver#FIELD_SEPARATOR}, removes that
+     * backslash. Needed when reading keys and resources from an SMS, before adding them to the
+     * dictionary.
+     *
+     * @param string The String where to remove every backslash preceding an occurrence of
+     *               {@link com.eis.smsnetwork.broadcast.BroadcastReceiver#FIELD_SEPARATOR}.
+     * @return A String where every backslash preceding an occurrence of
+     * {@link com.eis.smsnetwork.broadcast.BroadcastReceiver#FIELD_SEPARATOR} was removed.
+     */
+    public static String removeEscapes(@NonNull String string) {
+        String regex = Matcher.quoteReplacement("\\" + BroadcastReceiver.FIELD_SEPARATOR);
```

Comment:
i suggest
regex -->toReplace

Reply:
I don't know what you're talking about, could you explain?

Reply:
if you call toReplaceRegex or something with replace inside the String regex
by reading string.replaceAll(toReplaceRegex, BroadcastReceiver.FIELD_SEPARATOR);
I understand thet you replace the parts matching regex with Field-separator without knowing the order of parameter of replaceAll

Reply:
Yeah, I don't exactly know how that works, I didn't implement that myself. I'll send this suggestion to the correct person and I'll keep you updated 👍

Reply:
Yes, we should change the variable name to `toReplaceRegex`